src/user_works/handlers.clj

```
@@ -135,45 +136,18 @@
   (log/debug "Received:" message)
   {:status :ok})
 
-(def address-schema
-  {:street s/Str
-   (s/optional-key :street-2) s/Str
-   :city s/Str
-   :state s/Str
-   :zip s/Str
-   (s/optional-key :ocd-divisions)
-   #{address/division-regex}})
-
-(def email-schema
-  {:email s/Str
-   (s/optional-key :phone) s/Str})
-
-(def phone-schema
-  {:phone s/Str
-   (s/optional-key :email) s/Str})
-
-(def user-schema
-  {:first-name s/Str
-   :last-name s/Str
-   (s/optional-key :middle-name) s/Str
-   (s/optional-key :nickname) s/Str
-   (s/optional-key :addresses) {s/Keyword address-schema}
-   (s/optional-key :race) s/Keyword
-   (s/optional-key :dob) s/Inst
-   (s/optional-key :us-citizen) s/Bool
-   (s/optional-key :hostname) s/Str
-   (s/optional-key :language-preference) s/Keyword
-   (s/optional-key :referral-code) s/Str
-   (s/optional-key :party) s/Keyword
-   (s/optional-key :partner-comms-opt-in) s/Bool})
-
-(def create-user-schema
-  (s/conditional
-   #(not (str/blank? (:email %)))
-   (merge user-schema email-schema)
-
-   #(not (str/blank? (:phone %)))
-   (merge user-schema phone-schema)))
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; create                                                                   ;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(spec/def ::create-attrs (spec/keys :req-un [(or :user/email :user/phone)
+                                             :user/first-name :user/last-name]
+                                    :opt-un [:user/middle-name :user/nickname
+                                             :user/address :user/race :user/dob
+                                             :user/us-citizen :user/hostname
+                                             :user/referral-code :user/party
+                                             :user/language-preference
+                                             :user/partner-comms-opt-in]))
```

i went back and forth on this and i came down on running validations first because i wanted to (1) distinguish between bugs caused by invalid input and bugs in our namespacing code, and (2) fail fast on invalid input. i can see the value in namespacing first--especially since we plan on sending namespaced keys eventually anyway. i'll push up namespacing to step one.

@ddellacosta you're right. once we push up namespacing, the `:entity/user` spec will work just fine.